src/build123d/topology.py

```diff
@@ -379,26 +361,70 @@ def tuplify(obj: Any, dim: int) -> tuple:
     return result
 
 
+class _ClassMethodProxy:
+    """
+    A proxy for dynamically binding a class method to different classes.
+
+    This descriptor allows a class method defined in one class to be reused
+    in other classes while ensuring that the `cls` parameter refers to the
+    correct class (the class on which the method is being called). This avoids
+    issues where the method would otherwise always reference the original
+    defining class.
+
+    Attributes:
+        method (classmethod): The class method to be proxied.
+
+    Methods:
+        __get__(instance, owner):
+            Dynamically binds the proxied method to the calling class (`owner`).
+
+    Example:
+        class Mixin1D:
+            @classmethod
+            def extrude(cls, shape, direction):
+                print(f"extrude called on {cls.__name__}")
+
+        class Mixin2D:
+            extrude = ClassMethodProxy(Mixin1D.extrude)
+
+        class Mixin3D:
+            extrude = ClassMethodProxy(Mixin1D.extrude)
+
+        # Usage
+        Mixin2D.extrude(None, None)  # Output: extrude called on Mixin2D
+        Mixin3D.extrude(None, None)  # Output: extrude called on Mixin3D
+    """
+
+    def __init__(self, method):
+        self.method = method
+
+    def __get__(self, instance, owner):
+        # Bind the method dynamically as a class method of `owner`
+        return types.MethodType(self.method.__func__, owner)
+
+
 class Mixin1D:
     """Methods to add to the Edge and Wire classes"""
 
     def __add__(self, other: Union[list[Shape], Shape]) -> Self:
         """fuse shape to wire/edge operator +"""
 
-        # Convert `other` to list of base objects and filter out None values
+        # Convert `other` to list of base topods objects and filter out None values
         summands = [
             shape
             for o in (other if isinstance(other, (list, tuple)) else [other])
```

Comment:
```suggestion
            for o in ([other] if isinstance(other, (Shape)) else other)
```
reverse test to allow other to be `Iterable[Shape]`

Reply:
Done